src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -985,71 +1093,49 @@ private int constructMergedTable(Table table, List<FeedToMerge> feedsToMerge,
     }
 
-    private static String stopCodeFailureMessage(int stopsMissingStopCodeCount, int stopsCount, int specialStopsCount) {
-        return String.format(
-            "If stop_code is provided for some stops (for those with location_type = " +
-                "empty or 0), all stops must have stop_code values. The merge process " +
-                "found %d of %d total stops that were incorrectly missing stop_code values. " +
-                "Note: \"special\" stops with location_type > 0 need not specify this value " +
-                "(%d special stops found in feed).",
-            stopsMissingStopCodeCount,
-            stopsCount,
-            specialStopsCount
-        );
-    }
-
-    /** Get the set of shared fields for all feeds being merged for a specific table. */
-    private Set<Field> getSharedFields(List<FeedToMerge> feedsToMerge, Table table) throws IOException {
-        Set<Field> sharedFields = new HashSet<>();
-        // First, iterate over each feed to collect the shared fields that need to be output in the merged table.
-        for (FeedToMerge feed : feedsToMerge) {
-            CsvReader csvReader = table.getCsvReader(feed.zipFile, null);
-            // If csv reader is null, the table was not found in the zip file.
-            if (csvReader == null) {
-                continue;
-            }
-            // Get fields found from headers and add them to the shared fields set.
-            Field[] fieldsFoundInZip = table.getFieldsFromFieldHeaders(csvReader.getHeaders(), null);
-            sharedFields.addAll(Arrays.asList(fieldsFoundInZip));
-        }
-        return sharedFields;
-    }
-
     /**
-     * Checks whether a collection of fields contains a field with the provided name.
+     * Get the merge strategy to use for MTC service period merges by checking the active and future feeds for various
+     * combinations of matching trip and service IDs.
      */
-    private boolean containsField(Collection<Field> fields, String fieldName) {
-        for (Field field : fields) if (field.name.equals(fieldName)) return true;
-        return false;
-    }
-
-    /** Checks that any of a set of errors is of the type {@link NewGTFSErrorType#DUPLICATE_ID}. */
-    private boolean hasDuplicateError(Set<NewGTFSError> errors) {
-        for (NewGTFSError error : errors) {
-            if (error.errorType.equals(NewGTFSErrorType.DUPLICATE_ID)) return true;
+    private MergeStrategy getMergeStrategy(List<FeedToMerge> feedsToMerge) throws IOException {
+        // Iterate over both feeds to collect all trip and service IDs.
+        for (int i = 0; i < feedsToMerge.size(); i++) {
+            FeedToMerge feed = feedsToMerge.get(i);
+            Set<Table> tablesToCheck = Sets.newHashSet(Table.TRIPS, Table.CALENDAR, Table.CALENDAR_DATES);
+            for (Table table : tablesToCheck) {
+                feed.idsForTable.get(table).addAll(getIdsForTable(feed.zipFile, table));
+            }
```

Comment:
Can this be moved to inside class `FeedToMerge`?

Reply:
This only happens for service period merges, so I'm somewhat inclined to keep it here so that it has some amount of daylight.